File: server/src/main/java/com/backend/server/dto/SpaceResponse.java
package com.backend.server.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SpaceResponse {

    private Long id;
    private Long ownerId;
    private String address;
    private Integer squareFeet;
    private String vehicleTypes;
    private BigDecimal rentPerHour;
    private BigDecimal rentPerDay;
    private BigDecimal rentMonthly;
    @JsonProperty("imageUrls")
    private java.util.List<String> imageUrls;
    private LocalDateTime createdAt;
}

File: app/src/main/java/com/example/client/data/model/SpaceResponse.kt
package com.example.client.data.model

import com.google.gson.annotations.SerializedName

data class SpaceResponse(
    @SerializedName("id") val id: Long,
    @SerializedName("ownerId") val ownerId: Long,
    @SerializedName("address") val address: String,
    @SerializedName("squareFeet") val squareFeet: Int,
    @SerializedName("vehicleTypes") val vehicleTypes: String,
    @SerializedName("rentPerHour") val rentPerHour: Double? = null,
    @SerializedName("rentPerDay") val rentPerDay: Double? = null,
    @SerializedName("rentMonthly") val rentMonthly: Double? = null,
    @SerializedName("imageUrls") val imageUrls: List<String>? = null,
    @SerializedName("createdAt") val createdAt: String? = null
)
